docs/en/sql-reference/functions/string-search-functions.md

```diff
@@ -852,16 +852,42 @@ multiFuzzyMatchAllIndices(haystack, distance, \[pattern<sub>1</sub>, pattern<sub
 
 ## extract
 
-Extracts a fragment of a string using a regular expression. If `haystack` does not match the `pattern` regex, an empty string is returned.
-
-For regex without subpatterns, the function uses the fragment that matches the entire regex. Otherwise, it uses the fragment that matches the first subpattern.
+Extracts a fragment of a string using a regular expression.
+If `haystack` does not match the `pattern` regex, an empty string is returned. 
+For a regex without sub-patterns, the function uses the fragment that matches the entire regex. 
```

Comment:
L. 857 is kind of obvious. Instead of l. 857 and l. 858 we could have just a single sentence:

```md
If the regular expression has capturing groups, the function matches the input string against the first capturing group.
```